### PR TITLE
call SyncFlowCleanup during SetupFlow

### PR DESCRIPTION
**Pull request by heavycrystal**

### CODE REVIEW

**Review by serprex — APPROVED** [review]

File: flow/activities/flowable.go [hunk 1]
```diff
@@ -78,6 +78,13 @@ func (a *FlowableActivity) SetupMetadataTables(ctx context.Context, config *prot
 	}
 	defer connectors.CloseConnector(ctx, dstConn)
 
+	// this should have been done by DropFlowDestination
+	// but edge case due to late context cancellation
+	if err := dstConn.SyncFlowCleanup(ctx, config.FlowName); err != nil {
```
Comment (serprex) [inline flow/activities/flowable.go:83]: can e2e test this with a drop test that passes false for deleting stats

Comment (heavycrystal) [reply flow/activities/flowable.go:88]: can be done in #2463
